src/distilabel/steps/base.py

```diff
@@ -113,13 +113,18 @@ def model_post_init(self, __context: Any) -> None:
             self.pipeline = _GlobalPipelineManager.get_pipeline()
 
         if self.pipeline is None:
-            raise ValueError(
+            if not isinstance(self._logger, logging.Logger):
+                _logger = logging.getLogger(f"distilabel.step.{self.name}")
+            else:
+                _logger = self._logger
+            _logger.warning(
```

Comment:
What's the reason behind this change? Because AFAIK if the `self._logger` is initialized within the `load` call, it's never going to be something different than `logging.Logger`, right? i.e. the only scenario is whenever `self.load` is not called, right? cc @gabrielmbmb to confirm

Reply:
I've updated this to always get a logger at this stage (it wasn't throwing errors, but it was wrong anyway):
```python
            _logger = logging.getLogger(f"distilabel.step.{self.name}")
            _logger.warning(
                f"Step '{self.name}' hasn't received a pipeline, and it hasn't been"
                " created within a `Pipeline` context. Please, use"
                " `with Pipeline() as pipeline:` and create the step within the context."
            )
```
When this method runs we don't have yet the logger, and we want something to inform the user of the fact, but maybe the load method does something different than just loading the logger, so I think it's a good compromise, WDYT?